tests/models/dac/test_modeling_dac.py

```diff
@@ -0,0 +1,568 @@
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+
+            tuple_inputs = self._prepare_for_class(inputs_dict, model_class)
+            dict_inputs = self._prepare_for_class(inputs_dict, model_class)
+            check_equivalence(model, tuple_inputs, dict_inputs)
+
+    # Ignore copy
+    def test_initialization(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        configs_no_init = _config_zero_init(config)
+        for model_class in self.all_model_classes:
+            model = model_class(config=configs_no_init)
+            for name, param in model.named_parameters():
+                uniform_init_parms = ["conv", "in_proj", "out_proj", "codebook"]
+                if param.requires_grad:
+                    if any(x in name for x in uniform_init_parms):
+                        self.assertTrue(
+                            -1.0 <= ((param.data.mean() * 1e9).round() / 1e9).item() <= 1.0,
+                            msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                        )
+
+    def test_identity_shortcut(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs()
+        config.use_conv_shortcut = False
+        self.model_tester.create_and_check_model_forward(config, inputs_dict)
+
+
+def normalize(arr):
+    norm = np.linalg.norm(arr)
+    normalized_arr = arr / norm
+    return normalized_arr
+
+
+def compute_rmse(arr1, arr2):
+    arr1_normalized = normalize(arr1)
+    arr2_normalized = normalize(arr2)
+    return np.sqrt(((arr1_normalized - arr2_normalized) ** 2).mean())
+
+
+@slow
+@require_torch
+class DacIntegrationTest(unittest.TestCase):
+    def test_integration(self):
+        expected_rmse = {
+            "dac_16khz": 0.004,
+            "dac_24khz": 0.0026,
+            "dac_44khz": 0.0008,
+        }
+
+        expected_encoder_sums_dict = {
+            "dac_16khz": {
+                "loss": 24.873,
+                "quantized_representation": -22443.92,
+                "audio_codes": 1763635,
+                "projected_latents": 1891.828,
+            },
+            "dac_24khz": {
+                "loss": 28.091,
+                "quantized_representation": 7952.426,
+                "audio_codes": 7133234,
+                "projected_latents": -2933.110,
+            },
+            "dac_44khz": {
+                "loss": 24.1003,
+                "quantized_representation": 10457.930,
+                "audio_codes": 2282129,
+                "projected_latents": 2074.932,
+            },
+        }
+        librispeech_dummy = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+
+        models = ["dac_16khz", "dac_24khz", "dac_44khz"]
+
+        for model_name in models:
+            model_id = "kamilakesbi/{}".format(model_name)
```

Comment:
Remember to update to the Hub checkpoitns

Reply:
I'll update them as soon as we move them to the `descript` organisation!